gestalt-entity-system/src/main/java/org/terasology/gestalt/entitysystem/component/EmptyComponent.java

```diff
@@ -0,0 +1,16 @@
+package org.terasology.gestalt.entitysystem.component;
+
+/**
+ * An abstract component to be extended if a component that has no information to copy, like components that are flags
+ * for properties.
+ * <p>
+ * This should only be used for components that contain no information. If this is used for a component that does
+ * contain data, that data could be lost.
+ */
+public abstract class EmptyComponent implements Component<EmptyComponent> {
```

Comment:
I think it might be nicer to keep the generic on EmptyComponent partially bound (i.e.
```
public abstract class EmptyComponent<T> implements Component<T extends EmptyComponent>
```

This would ensure the bounds on the final component's copy method is correct.